src/DataTool.php

```diff
@@ -682,28 +694,14 @@ protected function truncateDataByLength($value, $length)
      */
     public function accessLocalField($fieldName)
     {
-        /* TODO - OPTIMIZATION - if we have to render the data,
-         * then save it to installData so we only do it once.
-         * Make sure that generateData checks for it.
-         * Note that this is safe even when used as a foreign
-         * object, becuase accessRemoteField calls clean each time.
-         */
         /* We will only deal with fields defined in the
          * vardefs.
          */
         if (!empty($this->fields[$fieldName])) {
-            /* If this data has already been generated,
-             * then just use it.
-             */
-            if (!empty($this->installData[$fieldName])) {
-                return $this->installData[$fieldName];
-                /* Otherwise, we have to pre-render it. */
-            } else {
-                $recSeed = $this->generateSeed($this->module, $fieldName, $this->count);
-                $recData = $this->fields[$fieldName];
-                $recType = (!empty($recData['dbType'])) ? $recData['dbType'] : $recData['type'];
-                return $this->getData($fieldName, $recType, $recData['type'], $recSeed);
+            if (empty($this->installData[$fieldName])) {
```

Comment:
Same here: `$this->installData[$fieldName]` can be equal `"0"`, in this case `empty("0")` returns true. Let's use `!isset` here.